libenv/unix_iface.h

```diff
@@ -0,0 +1,43 @@
+/*
+   Copyright (C) CFEngine AS
+
+   This file is part of CFEngine 3 - written and maintained by CFEngine AS.
+
+   This program is free software; you can redistribute it and/or modify it
+   under the terms of the GNU General Public License as published by the
+   Free Software Foundation; version 3.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+
+  To the extent this program is licensed as part of the Enterprise
+  versions of CFEngine, the applicable Commercial Open Source License
+  (COSL) may apply to this file if you as a licensee so wish it. See
+  included file COSL.txt.
+
+*/
+
+#ifndef CFENGINE_UNIX_IFACE_H
+#define CFENGINE_UNIX_IFACE_H
+
+
+#include <platform.h>
+
+/* TODO libpromises depends on libenv, the opposite should not happen! */
+#include <eval_context.h>
+
+
+void GetInterfacesInfo(EvalContext *ctx);
+void GetNetworkingInfo(EvalContext *ctx);
+
+typedef void (*ProcPostProcessFn)(void *ctx, void *json);
+ProcPostProcessFn NetworkingPortsPostProcessInfo(ARG_UNUSED void *passed_ctx, void *json);
+JsonElement* GetProcFileInfo(EvalContext *ctx, const char* filename, const char* key, const char* extracted_key, ProcPostProcessFn post, const char* regex);
+
+#endif
```

Comment:
I just removed this file, please don't bring it back! :-)

Either make the functions local to `unix_iface.c`, or put the function declarations in `sysinfo.h` instead.